### DISCUSSION TITLE
Installation on Mac with VS

### DISCUSSION THREAD

**Discussion by mijenner** [AvaloniaUI/Avalonia#7926] [Q&A]
Hints on using Avalonia with VS 2019 on Mac are very welcomed. 

First I ran: 

dotnet new -i Avalonia.Templates 

Next I manually created a folder 

mkdir MyFirstAvaMac 
cd MyFirstAvaMac 
dotnet new avalonia.mvvm 

Open project in VS 2019: 

First problem: .net 6 not installed, 
right click solution / project and set .net to 5. 

Running in debug mode seem to work somewhat. 

Two red marked files are: 
Avalonia.Themes.Default and
Avalonia.Themes.Fluent 

Should I worry about these? 

Please consider updating installation docs and add a section for VS.  

Kind regards, 
Michael

**Comment by maxkatz6** [comment]
> Two red marked files are

What does it mean?

**Reply by mijenner** [reply]
@maxkatz6 , It means the filenames are shown in solution explorer, but are marked red because they do not exist. 
I found references to them in .csproj e.g. 
`<TrimmableAssembly Include="Avalonia.Themes.Default" /> `

**Comment by robloo** [comment]
Avalonia now uses .NET 6. I believe this is known not to work well (if at all) with VS 2019 for macOS. I would highly suggest updating to the latest [VS 2022 for macOS previews](https://visualstudio.microsoft.com/vs/mac/preview/) to build Avalonia. I suspect things will work quite well once VS 2022 for mac is released (it's needed for .NET Maui as well).

**Reply by maxkatz6** [reply]
Unless they try to build iOS/Android projects, there shouldn't be any problems with .NET 5 and VS 2019.
At least from what I know.

**Reply by mijenner** [reply]
We only target Windows and MacOS and it seems to work ok, so far.
